lightning/src/ln/channel.rs

```diff
@@ -4569,7 +4619,7 @@ impl<Signer: Sign> Writeable for Channel<Signer> {
 
 		self.commitment_secrets.write(writer)?;
 
-		encode_varint_length_prefixed_tlv!(writer, {});
+		encode_varint_length_prefixed_tlv!(writer, {}, {(0, self.announcement_sigs)});
```

Comment:
Wondering why we don't give a length hint here and below?

Reply:
Maybe in a meeting we should discuss how we plan to support upgrading from various versions(?) e.g. should we support someone's upgrading from 0.0.15 to (in the future) 0.0.21 and all other possible permutations.

Reply:
The TLV writer currently doesn't support receiving length hints, we could add it in the future, but for now we only (really) do length hints during serialization of the `ln::msgs` structs, not the higher-level ones.